src/codecs/music_gme.c

```diff
@@ -0,0 +1,493 @@
+        FUNCTION_LOADER(gme_track_count, int (*)(Music_Emu const*))
+        FUNCTION_LOADER(gme_start_track, gme_err_t (*)( Music_Emu*,int))
+        FUNCTION_LOADER(gme_track_ended, int (*)( Music_Emu const*))
+        FUNCTION_LOADER(gme_set_tempo, void (*)(Music_Emu*,double))
+        FUNCTION_LOADER(gme_voice_count, int (*)(Music_Emu const*))
+        FUNCTION_LOADER(gme_mute_voice, void (*)(Music_Emu*,int,int))
+#if GME_VERSION >= 0x000700
+        FUNCTION_LOADER(gme_set_fade, void (*)(Music_Emu*,int,int))
+#else
+        FUNCTION_LOADER(gme_set_fade, void (*)(Music_Emu*,int))
+#endif
+        FUNCTION_LOADER(gme_track_info, gme_err_t (*)(Music_Emu const*, gme_info_t**, int))
+        FUNCTION_LOADER(gme_free_info, void (*)(gme_info_t*))
+        FUNCTION_LOADER(gme_seek, gme_err_t (*)(Music_Emu*,int))
+        FUNCTION_LOADER(gme_tell, int (*)(Music_Emu const*))
+        FUNCTION_LOADER(gme_play, gme_err_t (*)(Music_Emu*, int, short[]))
+        FUNCTION_LOADER(gme_delete, void (*)(Music_Emu*))
+#if defined(GME_DYNAMIC)
+        gme.gme_set_autoload_playback_limit = (void (*)(Music_Emu*,int)) SDL_LoadFunction(gme.handle, "gme_set_autoload_playback_limit");
+#elif defined(GME_HAS_SET_AUTOLOAD_PLAYBACK_LIMIT)
+        gme.gme_set_autoload_playback_limit = gme_set_autoload_playback_limit;
+#else
+        gme.gme_set_autoload_playback_limit = NULL;
+#endif
+
+#if defined(GME_DYNAMIC)
+        gme.gme_disable_echo = (void (*)(Music_Emu*,int)) SDL_LoadFunction(gme.handle, "gme_disable_echo");
+#elif defined(GME_HAS_DISABLE_ECHO)
+        gme.gme_disable_echo = gme_disable_echo;
+#else
+        gme.gme_disable_echo = NULL;
+#endif
+    }
+    ++gme.loaded;
+
+    return 0;
+}
+
+static void GME_Unload(void)
+{
+    if (gme.loaded == 0) {
+        return;
+    }
+    if (gme.loaded == 1) {
+#ifdef GME_DYNAMIC
+        SDL_UnloadObject(gme.handle);
+#endif
+    }
+    --gme.loaded;
+}
+
+/* This file supports Game Music Emulator music streams */
+typedef struct
+{
+    int play_count;
+    Music_Emu* game_emu;
+    int freesrc;
+    SDL_bool has_track_length;
+    int echo_disabled;
+    int track_length;
+    int intro_length;
+    int loop_length;
+    int volume;
+    double tempo;
+    double gain;
+    SDL_AudioStream *stream;
+    void *buffer;
+    size_t buffer_size;
+    Mix_MusicMetaTags tags;
+} GME_Music;
+
+void _Mix_GME_SetSpcEchoDisabled(void *music_p, int disabled)
+{
+    GME_Music *music = (GME_Music*)music_p;
+
+    if (music && gme.gme_disable_echo) {
+        gme.gme_disable_echo(music->game_emu, disabled);
+        music->echo_disabled = disabled;
+    }
+}
+
+int  _Mix_GME_GetSpcEchoDisabled(void *music_p)
+{
+    int ret = -1;
+    GME_Music *music = (GME_Music*)music_p;
+
+    if (music && gme.gme_disable_echo) {
+        ret = music->echo_disabled;
+    }
+
+    return ret;
+}
+
+static void GME_Delete(void *context);
+
+/* Set the volume for a GME stream */
+void GME_SetVolume(void *music_p, int volume)
```

Comment:
private procedures should be made static

Reply:
Thanks for the notice, I'll mark these calls being static at my fork side...